hypothesis-python/src/hypothesis/strategies/_internal/strings.py

```diff
@@ -331,15 +335,13 @@ def _identifier_characters():
     return id_start, id_continue
 
 
-class BytesStrategy(ListStrategy):
-    def __init__(self, min_size, max_size):
-        super().__init__(IntegersStrategy(0, 255), min_size=min_size, max_size=max_size)
+class BytesStrategy(SearchStrategy):
+    def __init__(self, min_size: int, max_size: Optional[int]):
+        self.min_size = min_size
+        self.max_size = max_size if max_size is not None else float("inf")
```

Comment:
I guess we have to assign _some_ size here to be compatible with the `ListStrategy` object, but a `float` is not an `int | None` and `data.draw_bytes()` expects the latter.

We could instead default to `BUFFER_SIZE` as the cap here?  Do we even need to accept `None` in the backend, or can we ensure that we always impose a finite maximum size between the user-facing API and the backend?

Reply:
ack, thanks for catching that clearly-wrong type. I think we can remove the nullability for max_size and set max_size to `COLLECTION_DEFAULT_MAX_SIZE` in the `None` case, both here and in strings – will try and see.